junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ExtensionRegistry.java

```diff
@@ -60,6 +62,10 @@
 	public static ExtensionRegistry createRegistryWithDefaultExtensions() {
 		ExtensionRegistry extensionRegistry = new ExtensionRegistry(null);
 		DEFAULT_EXTENSIONS.forEach(extensionRegistry::registerDefaultExtension);
+		if (Boolean.getBoolean(EXTENSIONS_AUTODETECT_PROPERTY_NAME)) {
```

Comment:
That method is invoked by `JupiterEngineDescriptor.before(JupiterEngineExecutionContext)`, and the `JupiterEngineExecutionContext` has access to the `ConfigurationParameters`.

So you'll need to update `JupiterEngineDescriptor` to invoke `createRegistryWithDefaultExtensions(context.getConfigurationParameters())`.